### PR TITLE
add swapr adaptor

### PR DESCRIPTION
**Pull request by mix1o**

### CODE REVIEW

**Comment by slasher125** [comment]
ty for the contribution! Looks good overall, though it seems its missing the token rewards from the `reward` field.
If you could add those on top that be great

**Comment by mix1o** [comment]
> ty for the contribution! Looks good overall, though it seems its missing the token rewards from the `reward` field. If you could add those on top that be great

not sure if I understand it, could you give an example ?

**Comment by slasher125** [comment]
Sure, lets take this pool as an example:

Screenshot 2022-05-26 at 10 58 09

here is the corresponding output from your adaptor:
```
  {
    pool: '0x1865d5445010e0baf8be2eb410d3eae4a68683c2',
    chain: 'Xdai',
    project: 'swapr',
    symbol: 'WETH-WXDAI',
    tvlUsd: 3321253.921630672,
    apy: 10.168893560444833
  },
```

So seems to me selected pools receive liquidity mining rewards on top of earned trading fees. On the reward tab we get more detailed info:

Screenshot 2022-05-26 at 11 04 19

The details about that can be found in the subgraph u are using already in the `rewards` field:

```
{
  address: '0x15e18d962f7be3071a945e3f89ddfad510da0d35',
  duration: '2419200',
  startsAt: '1649948400',
  endsAt: '1652367600',
  locked: false,
  stakingCap: '0',
  stakedAmount: '0',
  rewards: [
    {
      token: [Object],
      amount: '44800',
      __typename: 'LiquidityMiningCampaignReward'
    },
    {
      token: [Object],
      amount: '10.8',
      __typename: 'LiquidityMiningCampaignReward'
    },
    {
      token: [Object],
      amount: '10.8',
      __typename: 'LiquidityMiningCampaignReward'
    }
  ],
  stakablePair: {
    id: '0x1865d5445010e0baf8be2eb410d3eae4a68683c2',
    reserveNativeCurrency: '3363397.185884387506078818',
    reserveUSD: '3363397.185884387506078818',
    totalSupply: '35638.754512989043892815',
    reserve0: '899.570921857265633091',
    reserve1: '1681698.592942193753039409',
    token0: {
      address: '0x6a023ccd1ff6f2045c3309768ead9e68f978f6e1',
      name: 'Wrapped Ether on xDai',
      symbol: 'WETH',
      decimals: '18',
      __typename: 'Token'
    },
    token1: {
      address: '0xe91d153e0b41518a2ce8dd3d7944fa863463a97d',
      name: 'Wrapped XDAI',
      symbol: 'WXDAI',
      decimals: '18',
      __typename: 'Token'
    },
    __typename: 'Pair'
  },
  liquidityMiningPositions: [],
  __typename: 'LiquidityMiningCampaign'
}
```

So we'll need to calculate the liquidity mining rewards and add that on top of the fee apr. Makes sense?

**Comment by slasher125** [comment]
Basically, the adaptor so far is not wrong, just incomplete and users might wonder why its much lower than whats being displayed on the swapr UI

**Comment by mix1o** [comment]
> Basically, the adaptor so far is not wrong, just incomplete and users might wonder why its much lower than whats being displayed on the swapr UI

Thanks for help. I will make improvements

**Comment by slasher125** [comment]
yeah no worries, we are grateful for the contribution!

File: src/adaptors/swapr/index.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+const createPool = (pair, chainString) => {
+  const { id, token0, token1, totalValueLockedUSD: tvlUsd } = pair
+  
+  const symbol = utils.formatSymbol(
+    `${token0.symbol}-${token1.symbol}`
+  );
+    
+  const apy = rewardsAPR[id.toLowerCase()]
+  
+  const chain = utils.formatChain(chainString)
```
Comment (slasher125) [inline src/adaptors/swapr/index.js]: in case of chainString === 'xdai' lets use `xDai` as the chain

**Pull request merged**